src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs

```diff
@@ -550,14 +550,17 @@ private SyntaxToken RenameToken(SyntaxToken oldToken, SyntaxToken newToken, stri
                         }
                     }
                 }
-                else if (isAccessorLocation)
-                {
-                    var prefix = oldIdentifier.Substring(0, oldIdentifier.IndexOf('_') + 1);
-                    currentNewIdentifier = prefix + currentNewIdentifier;
-                }
-                else if (!string.IsNullOrEmpty(suffix))
+                else
                 {
-                    currentNewIdentifier = _replacementText + suffix;
+                    if (!string.IsNullOrEmpty(prefix))
```

Comment:
Is there a meaningful difference between prefix/suffix being null or being empty? Or can you just enforce they're always non-null, and then these ifs can go away since you're just concatenating empty strings?